pkg/engine/virtualmachinescalesets.go

```diff
@@ -417,7 +417,7 @@ func CreateAgentVMSS(cs *api.ContainerService, profile *api.AgentPoolProfile) Vi
 	addCustomTagsToVMScaleSets(profile.CustomVMTags, &virtualMachineScaleSet)
 
 	if profile.HasAvailabilityZones() {
-		virtualMachineScaleSet.Zones = &profile.AvailabilityZones
+		virtualMachineScaleSet.Zones = to.StringPtr(fmt.Sprintf("[parameters('%sAvailabilityZones')]", profile.Name))
```

Comment:
ditto above comment:

`&[]string{fmt.Sprintf("[parameters('%sAvailabilityZones')]", profile.Name)}

Reply:
fixed

Reply:
And for this one this can be the fix? `&[]string{fmt.Sprintf("[string(parameters('%sAvailabilityZones'))]", profile.Name)}`